### ISSUE TITLE
The generated HTML is not sematic

### ISSUE THREAD

**Issue by tirithen** [PolymerElements/paper-card#89]
When looking at the header for e.g. the paper-card component (https://elements.polymer-project.org/elements/paper-card) it does not use a h1 tag for the header. I would expect a section tag and a h1 in a card but all I can see are div tags. Is there a good reason for this or was it missed out on by accident?

**Comment by notwaldorf** [comment]
I don't think the header would be visible in the shadow DOM, so its semantics would be missed anyway. It's only semantically meaningful in shady DOM -- we could change that, but I'm not sure it's worth it given that the end goal is a world where shadow DOM is enabled everywhere. WDYT?

**Comment by tirithen** [comment]
Interesting point, but what happens with SEO in that case? The large search
engines does execute JavaScript so they will just see unstructured text? It
does feel weird and possibly harmful for your SEO to ignore HTML semantics,
although I guess it gets more complicated with Web Components

Den 5 dec. 2016 21:01 skrev "Monica Dinculescu" <notifications@github.com>:

**Comment by notwaldorf** [comment]
If SEO is a concern, you should distribute the content (much how like the contents of the card are distributed). Maybe the `header` shouldn't be an attribute, but a slot as well in that case.

**Comment by tirithen** [comment]
Yes exactly, I guess that the paper elements was not written with focus on
SEO and that I might have to create my own elements in the same style. Sure
would be nice though if this was part of the road map for the paper
framework I would maybe be able to get some hours from my work to
contribute on this, but if this is not at all your focus for any of the new
elements/refactorings I'm probably better off creating from scratch. That's
why I was wondering. I guess I should also read up a bit more on whether
the Google bot can render shadow dom properly although I would assume so
since it does run Javascript.

Den 7 dec. 2016 00:44 skrev "Monica Dinculescu" <notifications@github.com>:

**Comment by notwaldorf** [comment]
I entirely disagree that the paper elements were not written with SEO focus -- you are focusing on a very small bug, in the header of a card (where the rest of the content in the card is distributed, and therefore SEO parseable). You will notice that not many elements make this mistake, since most of them accept distributed content. Secondly, there is a very easy work around in the meantime, if this is an important issue to you -- you distribute a hidden header with the same content:

```
<paper-card heading="Card Title">
  <h1 hidden>Card Title</h1>
</paper-card>
```

Third, changing this inside the element itself shouldn't be very hard -- the div that displays the heading can take a `slot`, and if the slot is empty and the `heading` attribute is not, fallback to using the `heading`, so that it's still backwards compatible.

Please don't make generalizations about the entire "paper framework" based on one fairly small issue you've had with one element (which in all fairness nobody has raised in the year or so this element has been around), which is easily fixable.

**Comment by tirithen** [comment]
Thank you for the answer, will try not to make generalizations. Thanks for
the workaround and of course thanks for an otherwise really nice framework.

Den 7 dec. 2016 20:30 skrev "Monica Dinculescu" <notifications@github.com>:

I entirely disagree that the paper elements were not written with SEO focus
-- you are focusing on a very small bug, in the header of a card (where the
rest of the content in the card is distributed, and therefore SEO
parseable). You will notice that not many elements make this mistake, since
most of them accept distributed content. Secondly, there is a very easy
work around in the meantime, if this is an important issue to you -- you
distribute a hidden header with the same content:

<paper-card heading="Card Title">
  <h1 hidden>Card Title</h1>
</paper-card>

Third, changing this inside the element itself shouldn't be very hard --
the div that displays the heading can take a slot, and if the slot is empty
and the heading attribute is not, fallback to using the heading, so that
it's still backwards compatible.

Please don't make generalizations about the entire "paper framework" based
on one fairly small issue you've had with one element (which in all
fairness nobody has raised in the year or so this element has been around),
which is easily fixable.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/PolymerElements/paper-card/issues/89#issuecomment-265548047>,
or mute the thread
